plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerManager.kt

```diff
@@ -124,53 +126,64 @@ class CodeModernizerManager(private val project: Project) : PersistentStateCompo
         fun validateCore(project: Project): ValidationResult {
             if (isRunningOnRemoteBackend()) {
                 return ValidationResult(
                     false,
-                    message("codemodernizer.notification.warn.invalid_project.description.reason.remote_backend"),
                     InvalidTelemetryReason(
-                        CodeTransformPreValidationError.RemoteRunProject
+                        CodeTransformPreValidationError.RemoteRunProject,
                     )
                 )
             }
             if (!isCodeTransformAvailable(project)) {
                 return ValidationResult(
                     false,
-                    message("codemodernizer.notification.warn.invalid_project.description.reason.not_logged_in"),
                     InvalidTelemetryReason(
-                        CodeTransformPreValidationError.NonSsoLogin
+                        CodeTransformPreValidationError.NonSsoLogin,
                     )
                 )
             }
 
             if (ProjectRootManager.getInstance(project).contentRoots.isEmpty()) {
                 return ValidationResult(
                     false,
-                    message("codemodernizer.notification.warn.invalid_project.description.reason.missing_content_roots"),
                     InvalidTelemetryReason(
-                        CodeTransformPreValidationError.NoPom
+                        CodeTransformPreValidationError.EmptyProject,
                     )
                 )
             }
+
+            if (transformationType == CodeTransformType.SQL_CONVERSION) {
+                val javaModules = project.getJavaModules()
+                return if (javaModules.isNotEmpty()) {
+                    ValidationResult(
+                        true,
+                    )
+                } else {
+                    ValidationResult(
+                        false,
+                        InvalidTelemetryReason(
+                            CodeTransformPreValidationError.NoJavaProject,
+                        )
+                    )
+                }
+            }
+
             val supportedModules = project.getSupportedModules(supportedJavaMappings).toSet()
             val validProjectJdk = project.getSupportedJavaMappings(supportedJavaMappings).isNotEmpty()
             val projectJdk = project.tryGetJdk()
             if (supportedModules.isEmpty() && !validProjectJdk) {
                 return ValidationResult(
                     false,
-                    message(
-                        "codemodernizer.notification.warn.invalid_project.description.reason.invalid_jdk_versions",
-                        supportedJavaMappings.keys.joinToString()
-                    ),
                     InvalidTelemetryReason(
                         CodeTransformPreValidationError.UnsupportedJavaVersion,
-                        projectJdk?.toString().orEmpty()
+                        projectJdk.toString()
```

Comment:
Yes, it could be `null`, but from the `toString()` docs:

```
Returns a string representation of the object. Can be called with a null receiver, in which case it returns the string "null".
```

so this case is handled fine.